core/types/types.go

```diff
@@ -91,34 +91,34 @@ type Migration struct {
 	Timestamp        string `json:"timestamp"`          // Timestamp when the migration was proposed
 }
 
-type MigrationStatus int
+type MigrationStatus string
 
 const (
-	NoActiveMigration MigrationStatus = iota
-	MigrationNotStarted
-	MigrationInProgress
-	MigrationCompleted
-)
+	// NoActiveMigration indicates there is currently no migration process happening on the network.
+	NoActiveMigration MigrationStatus = "NoActiveMigration"
 
-func (status *MigrationStatus) String() string {
-	switch *status {
-	case NoActiveMigration:
-		return "NoActiveMigration"
-	case MigrationNotStarted:
-		return "MigrationNotStarted"
-	case MigrationInProgress:
-		return "MigrationInProgress"
-	case MigrationCompleted:
-		return "MigrationCompleted"
-	default:
-		return "Unknown"
-	}
-}
+	// ActivationPeriod represents the phase after the migration proposal has been approved by the network,
+	// but before the migration begins. During this phase, validators prepare their nodes for migration.
+	ActivationPeriod MigrationStatus = "ActivationPeriod"
+
+	// MigrationInProgress is the phase where the migration is actively occurring. The old and new networks
+	// run concurrently, with state changes from the old network being replicated to the new network.
+	MigrationInProgress MigrationStatus = "MigrationInProgress"
+
+	// MigrationCompleted indicates the migration process has successfully finished,
+	// and the old network is ready to be decommissioned.
+	MigrationCompleted MigrationStatus = "MigrationCompleted"
+
+	// GenesisMigration refers to the phase where the node initializes with the genesis state,
+	// tries to replicate the state changes from the old network.
+	GenesisMigration MigrationStatus = "GenesisMigration"
+)
 
 type MigrationState struct {
-	Status      MigrationStatus `json:"status"`       // Status is the current status of the migration
-	StartHeight int64           `json:"start_height"` // StartHeight is the block height at which the migration started
-	EndHeight   int64           `json:"end_height"`   // EndHeight is the block height at which the migration ends
+	Status       MigrationStatus `json:"status"`       // Status is the current status of the migration
+	StartHeight  int64           `json:"start_height"` // StartHeight is the block height at which the migration started
+	EndHeight    int64           `json:"end_height"`   // EndHeight is the block height at which the migration ends
+	CurrentBlock int64           `json:"chain_height"` // CurrentBlock is the current block height of the node
```

Comment:
Should doc that `CurrentBlock` will always be zero if status is requested from a node on the new network (`GenesisMigration`).  Also that this height pertains to height on the old network, because docs saying "current block height of the node" is not correct on new network nodes.

Also, why Current**Block**?  The others are "Height".  The json tag also says `chain_height`.

Reply:
Okay, I get the confusion around having this mixed up. Any ideas on how to seperate these 2 migration modes on the new and old?

Reply:
Especially the start and end height when we query for a node in genesis migration refers to the old chains migration start and end height. But if we start a new migration on the new chain, this reads different. These fields are rn very contextual.